examples/stress_tests/bevymark_3d.rs

```diff
@@ -0,0 +1,596 @@
+    #[argh(option, default = "1")]
+    material_texture_count: usize,
+
+    /// the alpha mode used to spawn the cubes
+    #[argh(option, default = "AlphaMode::Opaque")]
+    alpha_mode: AlphaMode,
+}
+
+#[derive(Default, Clone)]
+enum AlphaMode {
+    #[default]
+    Opaque,
+    Blend,
+    AlphaMask,
+}
+
+impl FromStr for AlphaMode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "opaque" => Ok(Self::Opaque),
+            "blend" => Ok(Self::Blend),
+            "alpha_mask" => Ok(Self::AlphaMask),
+            _ => Err(format!(
+                "Unknown alpha mode: '{s}', valid modes: 'opaque', 'blend', 'alpha_mask'"
+            )),
+        }
+    }
+}
+
+const FIXED_TIMESTEP: f32 = 0.2;
+
+fn main() {
+    // `from_env` panics on the web
+    #[cfg(not(target_arch = "wasm32"))]
+    let args: Args = argh::from_env();
+    #[cfg(target_arch = "wasm32")]
+    let args = Args::from_args(&[], &[]).unwrap();
+
+    App::new()
+        .add_plugins((
+            DefaultPlugins.set(WindowPlugin {
+                primary_window: Some(Window {
+                    title: "BevyMark 3D".into(),
+                    resolution: WindowResolution::new(1920, 1080).with_scale_factor_override(1.0),
+                    present_mode: PresentMode::AutoNoVsync,
+                    ..default()
+                }),
+                ..default()
+            }),
+            FrameTimeDiagnosticsPlugin::default(),
+            LogDiagnosticsPlugin::default(),
+        ))
+        .insert_resource(WinitSettings::continuous())
+        .insert_resource(args)
+        .insert_resource(BevyCounter {
+            count: 0,
+            color: Color::WHITE,
+        })
+        .add_systems(Startup, setup)
+        .add_systems(FixedUpdate, scheduled_spawner)
+        .add_systems(
+            Update,
+            (
+                mouse_handler,
+                camera_control_system,
+                movement_system,
+                collision_system,
+                counter_system,
+            ),
+        )
+        .insert_resource(Time::<Fixed>::from_duration(Duration::from_secs_f32(
+            FIXED_TIMESTEP,
+        )))
+        .run();
+}
+
+#[derive(Resource)]
+struct CubeScheduled {
+    waves: usize,
+    per_wave: usize,
+}
+
+fn scheduled_spawner(
+    mut commands: Commands,
+    args: Res<Args>,
+    mut scheduled: ResMut<CubeScheduled>,
+    mut counter: ResMut<BevyCounter>,
+    cube_resources: ResMut<CubeResources>,
+) {
+    if scheduled.waves > 0 {
+        let cube_resources = cube_resources.into_inner();
+        spawn_cubes(
+            &mut commands,
+            args.into_inner(),
+            &mut counter,
+            scheduled.per_wave,
+            cube_resources,
+            None,
+            scheduled.waves - 1,
+        );
+
+        scheduled.waves -= 1;
+    }
+}
+
+#[derive(Resource)]
+struct CubeResources {
+    _textures: Vec<Handle<Image>>,
```

Comment:
Why the underscore? It looks like the attribute is being used so it shouldn't warn about that?

Reply:
Ah, right, I momentarily forgot assignment wasn't enough to not trigger the warning.